src/classes/CountryStateJSonGenerator.cls

```diff
@@ -0,0 +1,186 @@
+        } catch (Exception e) {
+            System.debug('No Such Country which starts with '+ startingKeyword);
+        }
+        
+        // 2.Find the List of Districts whose length is greater than 
+        // or equals to Any specified Number
+        List<String> listOfDistrict = new List<String>();
+        for(District__c d : [SELECT Name FROM District__c]) {
+            if(d.Name.Length()>=12){
+                listOfDistrict.add(d.Name);
+            }
+        }
+        System.debug('Districts whose length greater than any no.: '+listOfDistrict);
+        
+		//Filling out the map dynamically
+        for(District__c dist : listDistrict) {
+            if(!mapCountryAndStateWithDistricts.containsKey(dist.State__r.Country__r.Name)){
+                mapCountryAndStateWithDistricts.put(dist.State__r.Country__r.Name, 
+                                                    new Map<String,List<String>>());
+            }
+            Map<String,List<String>> stateMap = 
+                mapCountryAndStateWithDistricts.get(dist.State__r.Country__r.Name);
+            if(!stateMap.containsKey(dist.State__r.Name)){
+                stateMap.put(dist.State__r.Name, new List<String>());
+            }
+            stateMap.get(dist.State__r.Name).add(dist.Name);   
+        }
+        // 3. We can also Find the List of Districts whose length is greater 
+        // than or equals to Any specified Number using map
+        /*
+        for(String gotCountry : mapCountryAndStateWithDistricts.keySet()) {
+            if(mapCountryAndStateWithDistricts.containsKey(gotCountry)) {
+                for(String gotState : mapCountryAndStateWithDistricts.get(gotCountry).keySet()) {
+                    if(mapCountryAndStateWithDistricts.get(gotCountry).containsKey(gotState)){
+                        for(String gotDistrict : 
+                            mapCountryAndStateWithDistricts.get(gotCountry).get(gotState)) {
+                                if(gotDistrict.length() >=12) {
+                                    System.debug('District : '+gotDistrict);
+                                } 
+                            }
+                    } 
+                } 
+            }
+        }
+		*/
+        System.debug(mapCountryAndStateWithDistricts.get('India'));
+        System.debug(mapCountryAndStateWithDistricts.get('United States'));
+        System.debug(mapCountryAndStateWithDistricts.get('Germany'));
+        System.debug(mapCountryAndStateWithDistricts.get('Switzerland'));   
+    }
+    
+    // Method for the JSON Generator
+    public void countryJSONGenerator(String countryName) {
+        Country wrapCountry = new country();
+        //for(String cont : mapCountryAndStateWithDistricts.keySet()){
+        wrapCountry.countryName = countryName;
+        States wrapState = new States();
+        wrapCountry.states = new List<States>();
+        
+        List<AggregateResult> listAggState = 
```

Comment:
do not give unnecessary line break and  follow the around 138 character rule
List<AggregateResult> listAggState = [Select COUNT(Name)Total_Number_Of_States FROM State__c 
                                                                 Where Country__r.Name LIKE :countryName 
                                                                 Group By Country__r.Name];